File: examples/undocumented/python_modular/preprocessor_pca_modular.py
#!/usr/bin/env python
from tools.load import LoadMatrix

lm=LoadMatrix()
data = lm.load_numbers('../data/fm_train_real.dat')

parameter_list = [[data]]

def preprocessor_pca_modular (data):
	from modshogun import RealFeatures
	from modshogun import PCA
	
	features = RealFeatures(data)
		
	preprocessor = PCA()
	preprocessor.init(features)
	preprocessor.apply_to_feature_matrix(features)

	return features


if __name__=='__main__':
	print('PCA')
	preprocessor_pca_modular(*parameter_list[0])

File: examples/undocumented/csharp_modular/preprocessor_pca_modular.cs
using System;

public class preprocessor_pca_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();
		double width = 1.4;
		int size_cache = 10;

		double[,] traindata_real = Load.load_numbers("../data/fm_train_real.dat");
		RealFeatures features = new RealFeatures(traindata_real);

		PCA preproc = new PCA();
		preproc.init(features);
		preproc.apply_to_feature_matrix(features);

	}
}